holidayextras/ui-toolkit
# A trial to add CSS Linting to the UI Toolkit

#### What does this PR do? (please provide any background)
- This PR creates a minified and full CSS file on build
- You can then run `npm run lint` on the Toolkit and it will lint both JS and CSS (on the un-minified file)
- It checks for bad coding behaviours in our styles. It does not check for bad LESS code, only the output.
- I have also made sure Travis works correctly and lints files on commit.
- Linting rules can be found in the `.csslintrc` files and notes on their use can be found at https://github.com/gruntjs/grunt-contrib-csslint#options

:note: this has been looked at by @hx-markterry 

#### What tests does this PR have?
This is a type of test all in itself

#### How can this be tested?
- First run `grunt docs` this should load up the UIToolkit locally - check it all looks pretty.
- Then in your terminal run `npm run lint` this will lint both the JS and CSS. It will fail on errors but just report on warnings (too many font sizes is one you may notice)
- Try changing some LESS... go on add `!important` I DARE you!
- You should now see linting fail and you should hold your head in shame for writing bad CSS.

#### Screenshots / Screencast
<img width="815" alt="screen shot 2015-09-09 at 16 28 54" src="https://cloud.githubusercontent.com/assets/778942/9766253/e544e8c2-570f-11e5-8016-773132d73ad9.png">

#### What gif best describes how you feel about this work?
![I DARE YOU](https://media.giphy.com/media/dnM2Q2E6nqE24/giphy.gif)

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):

#### Software Engineer or Developer review:

#### Software Engineer or project guru review:
